sycl/include/sycl/ext/intel/experimental/fpga_mem/properties.hpp

```diff
@@ -0,0 +1,323 @@
+//==----- properties.hpp - SYCL properties associated with fpga_mem ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/properties/property.hpp>       // for PropKind
+#include <sycl/ext/oneapi/properties/property_value.hpp> // for property_value
+
+#include <cstdint>     // for uint16_t
+#include <iosfwd>      // for nullptr_t
+#include <string_view> // for string_view
+#include <type_traits> // for true_type
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+// Forward declare a class that these properties can be applied to
+template <typename T, typename PropertyListT> class fpga_mem;
+
+// Make sure that we are using the right namespace
+template <typename PropertyT, typename... Ts>
+using property_value =
+    sycl::ext::oneapi::experimental::property_value<PropertyT, Ts...>;
+
+// Property definitions
+enum class resource_enum : std::uint16_t { mlab, block_ram };
+
+struct resource_key {
+  template <resource_enum Resource>
+  using value_t =
+      property_value<resource_key,
+                     std::integral_constant<resource_enum, Resource>>;
+};
+
+struct num_banks_key {
+  template <size_t elements>
+  using value_t =
+      property_value<num_banks_key, std::integral_constant<size_t, elements>>;
+};
+
+struct stride_size_key {
+  template <size_t elements>
+  using value_t =
+      property_value<stride_size_key, std::integral_constant<size_t, elements>>;
+};
+
+struct word_size_key {
+  template <size_t elements>
+  using value_t =
+      property_value<word_size_key, std::integral_constant<size_t, elements>>;
+};
+
+struct bi_directional_ports_key {
+  template <bool Enable>
+  using value_t =
+      property_value<bi_directional_ports_key, std::bool_constant<Enable>>;
+};
+
+struct clock_2x_key {
+  template <bool Enable>
+  using value_t = property_value<clock_2x_key, std::bool_constant<Enable>>;
+};
+
+enum class ram_stitching_enum : std::uint16_t { min_ram, max_fmax };
+
+struct ram_stitching_key {
+  template <ram_stitching_enum Ram_stritching>
+  using value_t = property_value<
+      ram_stitching_key,
+      std::integral_constant<ram_stitching_enum, Ram_stritching>>;
+};
+
+struct max_private_copies_key {
+  template <size_t n>
```

Comment:
```suggestion
  template <size_t N>
```
Or alternatively some more descriptive names for these template arguments. At the very least follow the style of the naming.

The same goes for all the other template arguments here that use lowercase names.